src/commonTest/kotlin/app/cash/turbine/FlowTurbineScopeTest.kt

```diff
@@ -0,0 +1,125 @@
+import kotlinx.coroutines.delay
+import kotlinx.coroutines.flow.emitAll
+import kotlinx.coroutines.flow.emptyFlow
+import kotlinx.coroutines.flow.flow
+import kotlinx.coroutines.flow.flowOf
+import kotlinx.coroutines.flow.onCompletion
+import kotlinx.coroutines.flow.onStart
+import kotlinx.coroutines.launch
+import kotlinx.coroutines.test.runTest
+import kotlinx.coroutines.withContext
+
+class FlowTurbineScopeTest {
+  @Test fun multipleFlows() = runTest {
+    val turbine1 = flowOf(1).testIn(this)
+    val turbine2 = flowOf(2).testIn(this)
+    assertEquals(1, turbine1.awaitItem())
+    assertEquals(2, turbine2.awaitItem())
+    turbine1.awaitComplete()
+    turbine2.awaitComplete()
+  }
+
+  @Test fun cancelMustBeCalled() = runTest {
+    val job = launch {
+      coroutineScope {
+        neverFlow().testIn(this)
+      }
+    }
+    // Wait on real dispatcher for wall clock time. This almost certainly means we'd wait forever.
+    withContext(Default) {
+      delay(1.seconds)
+    }
+    job.cancel()
+  }
+
+  @Test fun cancelStopsFlowCollection() = runTest {
+    var collecting = false
+    val turbine = neverFlow()
+      .onStart { collecting = true }
+      .onCompletion { collecting = false }
+      .testIn(this)
+
+    assertTrue(collecting)
+    turbine.cancel()
+    assertFalse(collecting)
+  }
+
+  @Test fun unconsumedItemThrows() {
+    val actual = assertFailsWith<CompletionHandlerException> {
+      runTest {
+        flow {
+          emit("item!")
+          emitAll(neverFlow()) // Avoid emitting complete
+        }.testIn(this).cancel()
+      }
+    }
+    val cause = actual.cause
+    assertTrue(cause is AssertionError)
+    assertEquals(
+      """
+      |Unconsumed events found:
+      | - Item(item!)
+      """.trimMargin(),
+      cause.message,
+    )
+  }
+
+  @Test fun unconsumedCompleteThrows() {
+    val actual = assertFailsWith<CompletionHandlerException> {
+      runTest {
+        emptyFlow<Nothing>().testIn(this)
+      }
+    }
+    val cause = actual.cause
+    assertTrue(cause is AssertionError)
+    assertEquals(
+      """
+      |Unconsumed events found:
+      | - Complete
+      """.trimMargin(),
+      cause.message,
+    )
+  }
+
+  @Test fun unconsumedErrorThrows() {
+    val expected = RuntimeException()
+    val actual = assertFailsWith<CompletionHandlerException> {
+      runTest {
+        flow<Nothing> { throw expected }.testIn(this)
+      }
+    }
+    val cause = actual.cause
+    assertTrue(cause is AssertionError)
+    assertEquals(
+      """
+      |Unconsumed events found:
+      | - Error(RuntimeException)
+      """.trimMargin(),
+      cause.message,
+    )
+    assertSame(expected, cause.cause)
+  }
+
+  @Test fun unconsumedEventNotIncludedOnException() {
+    val expected = RuntimeException("hey")
+    val actual = assertFailsWith<RuntimeException> {
+      runTest {
+        flowOf(1).testIn(this)
+        throw expected
+      }
+    }
+    assertSame(expected, actual)
```

Comment:
Missing `assertNull(actual.cause)` and `assertEquals(emptyList(), actual.suppressed)`